spec/integration/user_show_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+RSpec.describe 'User Show Page', type: :feature do
+  before do
+    @user = User.create(id: 100, name: 'Me', photo: 'img1.jpg', bio: 'A teacher', posts_counter: 1)
+    @user.save
+
+    @post = Post.create(title: 'Post 1', text: 'My first post', author: @user, comments_counter: 2, likes_counter: 1)
+
+    visit user_path(@user)
+  end
+
+  describe 'User information' do
+    it 'renders the profile picture' do
+      expect(page).to have_css("img[src*='#{@user.photo}']")
+    end
+
+    it 'renders the username' do
+      expect(page).to have_content(@user.name)
+    end
+
+    it 'renders the number of posts' do
+      expect(page).to have_content("Number of posts: #{@user.posts_counter}")
+    end
+
+    it "renders user's bio" do
+      expect(page).to have_content(@user.bio)
+    end
+
+    it 'renders the first 3 posts of the user' do
+      expect(page).to have_content(@post.title)
+    end
```

Comment:
- [ ] This test requires that there should be up to 3 posts. For a more accurate assertion, consider creating more posts not just one.